### PR TITLE
Make S1U/N3 address config optional

### PR DESCRIPTION
**Pull request by daniele-moro**
Also, change all instances of S1U to N3 address to be coherent with 5G terminology.

Depends on:
- [x] https://github.com/omec-project/upf-epc/pull/468

### CODE REVIEW

File: app/app/src/main/java/org/omecproject/up4/impl/Up4DeviceManager.java [hunk 1]
```diff
@@ -412,7 +412,9 @@ private void ensureInterfacesInstalled() {
     @Override
     public Collection<UpfInterface> configInterfaces() {
         Collection<UpfInterface> interfaces = new ArrayList<>();
-        interfaces.add(UpfInterface.createS1uFrom(config.s1uAddress(), SLICE_MOBILE));
+        if (config.n3Address().isPresent()) {
+            interfaces.add(UpfInterface.createS1uFrom(config.n3Address().get(), SLICE_MOBILE));
```
[the remark this answers is not in the document]
Comment (osinstom) [reply app/app/src/main/java/org/omecproject/up4/impl/Up4DeviceManager.java:416]: I guess it has to be changed in the UpfProgrammable API? If so, it's ok to ignore this for now to speed up the process.

[the remark this answers is not in the document]
Comment (daniele-moro) [reply app/app/src/main/java/org/omecproject/up4/impl/Up4DeviceManager.java:416]: If we merge this first, I'll change it: https://gerrit.onosproject.org/c/onos/+/25285

**Comment by daniele-moro** [comment]
retest this please

**Comment by daniele-moro** [comment]
retest this please

File: app/app/src/main/java/org/omecproject/up4/impl/Up4DeviceManager.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (pierventre) [reply app/app/src/main/java/org/omecproject/up4/impl/Up4DeviceManager.java:416]: let's do it then

File: scenarios/p4rt-ifaces-config.xml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+<!--
+ ~ SPDX-License-Identifier: Apache-2.0
+ ~ SPDX-FileCopyrightText: 2022-present Open Networking Foundation <info@opennetworking.org>
+  -->
+<scenario name="p4rt-ifaces-config" description="Configure UE pool on the UP4 interfaces table">
+    <group name="P4rt-Ifaces-Config">
+        <step name="P4rt-Ifaces-Config-Push-Uepool"
+              exec="${DOCKER_COMPOSE_CMD} exec -T p4rt p4rt-up4-flows program --add-iface-uepool --ue-pool 17.0.0.0/16 --ue-count 0 --election-id 105"/>
+        <step name="P4rt-Ifaces-Config-Check-Uepool" requires="^"
+              exec="onos-cli-grep ${OCI} up4:read-interfaces 'UpfInterface(type=CORE, prefix=17.0.0.0/16'"/>
```
Comment (pierventre) [inline scenarios/p4rt-ifaces-config.xml:10]: I remember in some files /24. Do we need to update elsewhere ?

**Review by pierventre — APPROVED** [review]

File: scenarios/p4rt-ifaces-config.xml (hunk 2 above)
Comment (daniele-moro) [reply scenarios/p4rt-ifaces-config.xml:10]: No that's ok. The `/24` you see in pfcp/p4rt scenarios is to be able to generate non-overlapping UE IPs when we have multiple eNBs (i.e., for eNB 1 we generate UE IP address into the 17.0.0.0/24 subnet, while for eNB2 17.0.1.0/24).

**Comment by daniele-moro** [comment]
retest this please

**Pull request merged**